src/opentracing/span_context.js

```diff
@@ -9,7 +9,16 @@ class DatadogSpanContext extends SpanContext {
     this.traceId = props.traceId
     this.spanId = props.spanId
     this.parentId = props.parentId || null
-    this.sampled = props.sampled === undefined || props.sampled
+    if (props.samplingPriority !== undefined) {
+      this.samplingPriority = props.samplingPriority
+    }
+    if (props.sampled !== undefined) {
+      this.sampled = props.sampled
+    } else if (this.samplingPriority !== undefined) {
+      this.sampled = this.samplingPriority > 0
```

Comment:
I would leave `sampled` independent of sampling priority. The reasoning is that client sampling is additional sampling that can be done on top of sampling priority. Both functionalities are not really related and can work independently.

`sampled` should always be undefined (and this true) when there is a parent anyway. If client sampling is not wanted at all, just leaving the default value will keep everything at the client level.

Reply:
Okay, makes sense they are unrelated; I really wasn't sure on that...

The previous code was doing `this.sampled = props.sampled === undefined || props.sample` in span context's constructor...was that right? That was part of my confusion here, e.g. I had a fallback to set `sampled=true` when `props.sampled` was `undefined`.

Reply:
I put back the previous `props.sampled` line and pushed the various fixes. Thanks!